### PR TITLE
Add schema-based rules for pluggable monitor dependencies data of package index

### PR DESCRIPTION
**Pull request by per1234**
These are the JSON schema-based rules to check the [`packages[].platforms[].monitorDependencies[]`-level data](https://arduino.github.io/arduino-cli/dev/package_index_json-specification/#platforms-definitions) of Arduino Package Index projects.

### CODE REVIEW

File: internal/project/projectdata/testdata/packageindexes/empty-ids/package_foo_index.json [hunk 1]
```diff
@@ -43,6 +43,16 @@
               "packager": "barpackager",
               "name": "carrier-pigeon-discovery"
             }
+          ],
+          "monitorDependencies": [
+            {
+              "packager": "arduino",
+              "name": "networ-monitor"
```
Comment (silvanocerza) [inline internal/project/projectdata/testdata/packageindexes/empty-ids/package_foo_index.json:50]:
```suggestion
              "name": "network-monitor"
```

**Review by silvanocerza — CHANGES_REQUESTED** [review]
Some small typos need to be fixed but the rest looks great. :+1:

File: internal/project/projectdata/testdata/packageindexes/empty-ids/package_foo_index.json (hunk 1 above)
Comment (per1234) [reply internal/project/projectdata/testdata/packageindexes/empty-ids/package_foo_index.json]: Fixed. Thanks!
https://github.com/arduino/arduino-lint/blob/ec2851f01e9eaa88324e35061f49968d9a7791ea/internal/project/projectdata/testdata/packageindexes/empty-ids/package_foo_index.json#L50

File: internal/rule/ruleconfiguration/ruleconfiguration.go [hunk 2]
```diff
@@ -4700,6 +4700,142 @@ var configurations = []Type{
...
+		WarningModes:     nil,
+		ErrorModes:       []rulemode.Type{rulemode.Default},
+		RuleFunction:     rulefunction.PackageIndexPackagesPlatformsMonitorDependenciesPackagerLTMinLength,
+	},
+	{
+		ProjectType:      projecttype.PackageIndex,
+		SuperprojectType: projecttype.All,
+		Category:         "data",
+		Subcategory:      "platform",
+		ID:               "IL72",
```
[the remark this answers is not in the document]
Comment (per1234) [reply internal/rule/ruleconfiguration/ruleconfiguration.go]: Fixed. Thanks!
https://github.com/arduino/arduino-lint/blob/ec2851f01e9eaa88324e35061f49968d9a7791ea/internal/rule/ruleconfiguration/ruleconfiguration.go#L4793

**Review by silvanocerza — APPROVED** [review]

**Pull request merged**